projects/core/src/occ/utils/occ-fields.spec.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  extractFields,
+  mergeFields,
+  optimizeFields,
+  parseFields,
+  stringifyFields,
+} from './occ-fields';
+
+describe('mergeFields', () => {
+  it('should merge simple fields', () => {
+    expect(mergeFields(['ala', 'ma'])).toEqual('ala,ma');
+    expect(mergeFields(['ala,ula', 'ma'])).toEqual('ala,ula,ma');
+    expect(mergeFields(['ala,ula', 'ma,kota'])).toEqual('ala,ula,ma,kota');
+    expect(mergeFields(['ala,ula', 'ma,kota,i', 'psa'])).toEqual(
+      'ala,ula,ma,kota,i,psa'
+    );
+    expect(mergeFields(['ala,ula', 'ma,kota,i', 'psa'])).toEqual(
+      'ala,ula,ma,kota,i,psa'
+    );
+
+    expect(mergeFields(['ala,ula', 'ala,ela,ma,kota,i', 'kota,psa'])).toEqual(
+      'ala,ula,ela,ma,kota,i,psa'
+    );
+  });
+
+  it('should merge nested fields', () => {
+    expect(mergeFields(['ala(lubi)', 'ma'])).toEqual('ala(lubi),ma');
+    expect(mergeFields(['ala(lubi),ula', 'ma,ala(chodzi)'])).toEqual(
+      'ala(lubi,chodzi),ula,ma'
+    );
+    expect(
+      mergeFields(['ala(lubi(koty)),ula', 'ma,ala(lubi(i,psy),chodzi,pieszo)'])
+    ).toEqual('ala(lubi(koty,i,psy),chodzi,pieszo),ula,ma');
+  });
+
+  it('should optimize BASIC, DEFAULT and FULL fields', () => {
+    expect(mergeFields(['test(FULL)', 'test(DEFAULT)', 'test(BASIC)'])).toEqual(
```

Comment:
Can we add unit tests for special case: DEFAULT 'beats' BASIC

Reply:
this case is covered in `optimizeFields`.